src/app/attendance/components/AttendanceFormBuilder.tsx

```diff
@@ -44,74 +48,56 @@ const attendanceSchema = z
     }
   );
 export type AttendanceFormValues = z.infer<typeof attendanceSchema>;
-const loginUser: User = {
-  id: 1,
-  name: "Alice Johnson",
-  shift: "A",
-  role: "incharge",
-  status: UserStatus.Active,
-  phone: "123-456-7890",
-  address: "123 Main St, Anytown, USA",
-  created_at: new Date().toISOString(),
-  updated_at: new Date().toISOString(),
-  deleted_at: null,
-};
 
-const AttendanceFormBuilder = () => {
+const AttendanceFormBuilder: React.FC<AttendanceFormBuilderProps> = ({
+  initialData,
+  loginUser,
+}) => {
   const form = useForm<AttendanceFormValues>({
     resolver: zodResolver(attendanceSchema),
     defaultValues: {
       memberId:
-        loginUser.role === MemberRole.Member
+        loginUser?.role === MemberRole.Member
           ? loginUser.id.toString()
-          : undefined,
-      startTime: "",
-      endTime: "",
+          : initialData?.memberId || undefined,
+      startTime: initialData?.startTime ?? "",
+      endTime: initialData?.endTime || "",
     },
   });
-  const params = useParams();
-  const paramId = params?.id;
-  useEffect(() => {
-    const fetchAttendance = async () => {
-      try {
-        if (paramId) {
-          const response = await getAttendanceById(Number(paramId));
 
-          if (response) {
-            const { memberId, startTime, endTime } = response.data;
-            form.reset({
-              memberId: memberId.toString(),
-              startTime,
-              endTime,
-            });
-          } else {
-            toast({
-              title: "Attendance not found",
-              description: "No data found for this ID",
-            });
-          }
-        }
-      } catch (error) {
-        if (error instanceof Error) {
+  const onSubmit = async (values: AttendanceFormValues) => {
+    if (!loginUser?.id) {
+      toast({
+        title: "Error",
+        description: "User ID is missing.",
+      });
+      return;
+    }
+
+    const updatedValue = {
+      id: loginUser.id.toString(),
```

Comment:
I don't think we need this. Isn't id the id of the attendance? Why do we replace it with user id?

Reply:
Fixed